lib/system_description.rb

```diff
@@ -218,11 +218,22 @@ def scopes
   end
 
   def assert_scopes(*scopes)
-    missing = scopes.select { |scope| !self[scope] || self[scope].empty? }
+    missing_scopes = scopes.select { |scope| !self[scope] }
 
-    unless missing.empty?
+    if !missing_scopes.empty?
       raise Machinery::Errors::SystemDescriptionError.new(
-        "The system description misses the following section(s): #{missing.join(", ")}."
+        "The system description misses the following" \
+          " #{Machinery.pluralize(missing_scopes.size, "scope")}: #{missing_scopes.join(", ")}."
+      )
+    end
+
+    empty_scopes = scopes.select { |scope| !self[scope] || self[scope].empty? }
```

Comment:
The `!self[scope]` case is already covered in line 221 so it doesn't have any effect here anymore. I'd remove it from the clause.